### PR TITLE
ft(AddToWishlist): user add product to wishlist

### PR DESCRIPTION
**Pull request by martinemahirwe**
created the component for add product to wishlist functionality added functionality
wrote tests for testing the functionality for adding product to wishlist [delivers #187364828]

## ℹ️ Description

Please write about the changes implemented in this PR. Also write a short description about why the change is implemented.

## 🧪 How can this be tested?

Write a short description on how others can test the changes implemented in the PR

## 📸 Screenshots
Screenshot from 2024-07-16 09-53-56

wishlist modal
Screenshot from 2024-07-16 09-50-38

## 📍 Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Aldot-02 — DISMISSED** [review]
LGTM

File: src/components/rootcomponents/AddToWishlist.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  className?: string;
+  color?: string;
+}> = ({
+  productId,
+  isInWishlist: initialIsInWishlist,
+  className,
+  color = 'currentColor',
+}) => {
+  const [isInWishlist, setIsInWishlist] = useState(initialIsInWishlist);
+  const [addProductToWishlist, { isLoading, isSuccess, isError }] =
```
Comment (shebz2023) [inline src/components/rootcomponents/AddToWishlist.tsx]: const [addProductToWishlist, { isLoading, isSuccess, isError ,error}]

File: src/components/rootcomponents/AddToWishlist.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  useEffect(() => {
+    if (isSuccess) {
+      setIsInWishlist(true);
+      toast.success('Product added to wishlist successfully.');
+    }
+
+    if (isError) {
+      toast.error(
+        'Failed to add product to wishlist. Please login and try again.',
```
Comment (shebz2023) [inline src/components/rootcomponents/AddToWishlist.tsx]: toast.error(
        'Failed to add product to wishlist.' + error )

**Review by Oliviier-dev — DISMISSED** [review]
Nice work, @martinemahirwe , LGTM

**Review by shebz2023 — COMMENTED** [review]
Good work so far @martinemahirwe , some suggestions bellow, and its better to navigate users to login page when not logged in

**Review by shebz2023 — DISMISSED** [review]
LGTM

**Review by Oliviier-dev — DISMISSED** [review]
LGTM

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
1. Use a noticeable color for the wishlist button inside the image, you can also add some shadow around for it to be noticeable
2. Improve the UI design for the wishlist modal, You can find more inspiration online
3. Rebase. 

@martinemahirwe

**Review by Oliviier-dev — APPROVED** [review]
LGTM

**Review by shebz2023 — APPROVED** [review]
LGTM

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**